canonical/maas-ui
# feat(vault): Add vault setup instructions to settings

## Done

- Add MAAS Vault docs to docsUrls (not live)
- Add vault setup instructions to security settings page
- Separate security settings into "Security protocols" and "Secret storage"
- Add new custom icons for vault status

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /settings/configuration/security
- Ensure the Vault setup instructions are displayed without warning messages (see first screenshot)

## Fixes

Fixes https://github.com/canonical/app-tribe/issues/1410 
Fixes https://github.com/canonical/app-tribe/issues/1409
Fixes https://github.com/canonical/app-tribe/issues/1408
Fixes https://github.com/canonical/app-tribe/issues/1412

## Screenshots

### Before Vault setup
![image](https://user-images.githubusercontent.com/35104482/196649202-a47ad0fc-9f04-47f8-aded-762d287cc519.png)

### Vault partially set up
![image](https://user-images.githubusercontent.com/35104482/196649374-911c1c93-b7ae-4293-8724-f99472b37522.png)

### All controllers configured but secrets not migrated
![image](https://user-images.githubusercontent.com/35104482/196649716-8f98a8a6-a09f-49d5-b779-c15a5e6270df.png)

### Vault configuration complete
![image](https://user-images.githubusercontent.com/35104482/196649934-3e34b1bc-9cb1-4c2d-9fdb-4da15003230f.png)
